### DISCUSSION TITLE
Saving interactive plots or a Convenient way to Instantiate Document class from Exported Results?

### DISCUSSION THREAD

**Discussion by ml-l** [mindee/doctr#1672] [General]
Hi, I'm new to docTR and I'm trying to run a trial to see qualitatively, how well a model performs on a dataset I have.

Is there a way to save the interactive plot that the `doctr.utils.visualization.visualize_page` function makes in a more platform agnostic manner without losing the interactivity from the mplcursors (e.g. something like how plotly can save figures as html and still be interactive)? I know I can probably pickle dump the created figure, but I'd prefer an option that's more transparent and more shareable.

Or alternatively, is there a convenient way to instantiate the `Document` or `Page` class from a saved JSON result that we get from the `.export()` method from said classes?

**Comment by felixdittrich92** [comment]
Hi @ml-l  :wave:,

Excuse the late response.

You can instantiate the single elements from the json-like dictonary yes

See:
https://github.com/mindee/doctr/blob/9045dcfc9c5c837b06fcda8e802f7cf1d95bd18c/doctr/io/elements.py#L426

For example:
```
# from dict
state_dict = {
    "words": [
        {
            "value": "there",
            "confidence": 0.1,
            "geometry": ((0, 0), (1.0, 1.0)),
            "objectness_score": objectness_score,
            "crop_orientation": {"value": 0, "confidence": None},
        }
    ],
    "geometry": ((0, 0), (1.0, 1.0)),
    "objectness_score": objectness_score,
}
line = elements.Line.from_dict(state_dict)
```
